test/cql-pytest/test_wasm.py

```diff
@@ -990,11 +992,15 @@ def test_UDA(cql, test_keyspace, table1, scylla_with_wasm_only, metrics):
           res = [row for row in cql.execute(f"SELECT i2 AS result FROM {table}")]
 
           misses_after_reuse = get_metric(metrics, 'scylla_user_functions_cache_misses')
+          # Sum of hits and misses should equal the total number of UDF calls, which is one row function call for
+          # each of the table elements and one additional call for the final function, both multiplied by the number of repetitions.
           assert misses_after_reuse - misses_before_reuse + get_metric(metrics, 'scylla_user_functions_cache_hits') - hits_after == 100 * (1 + len(res))
-          # each shard has its own cache, so check if at least one shard reuses the cache without missing (uda combines 2 functions)
-          # taking into account misses caused by first use of the uda in a shard and misses caused by too big of a growth of an instance
-          # that was used multiple times, we estimate that happens once every 15 uses
-          assert misses_after_reuse - misses_before_reuse < 200 + 100 * (len(res)) / 15
+          # Each shard has its own cache, so check if at least one shard reuses the cache without excessive missing.
+          # Misses caused by replacing an instance that can no longer be used can be justified, we estimate that this happens
+          # once every 7 calls for row function calls (memory has 2 initial pages, 2 pages are added for 2 arguments, 16 pages = 1MB)
+          # and once every 14 calls for final function calls (1 page is added for each call in this case). Additionally, 2 misses are
+          # expected for each of the 2 shards, for the first calls of row and final functions
```

Comment:
This rationale misses an important bit of context that ScyllaDB currently reserved 1MB for the cache, I presume? That's the only constant that was not initially obvious to me when trying to parse the calculations

Reply:
Yes, I'll add this in a second